File: core/weapons/heavyRailGun.js
import { worldSize } from "../utils/constants.js";
import { wrap } from "../world/utils.js";
import Projectile from "./projectile.js";

export default class HeavyRailGun extends Projectile {
    constructor({ x, y, angle, ship, color, speed = 10 }) {
        super({
            name: "Heavy RailGun",
            speed: speed * 4,
            acceleration: 50000,
            x: x,
            y: y,
            width: 20,
            height: 90,
            angle: angle,
            damage: 100,
            range: 50000,
            fireRate: 0.005,
            energyCost: 1000,
            penetration: 5,
            ship,
            color
        });
    }

    // update(dt, manager) {
    //     super.update(dt, manager)
    // }
}

File: void_front/game/core/weapons/heavy_rail_gun.py
from __future__ import annotations

from game.core.weapons.projectile import Projectile


class HeavyRailGun(Projectile):
    def __init__(self, options, force=False):
        defaults = {
            "name": "Heavy RailGun",
            "speed": options.get("speed", 10) * 4,
            "acceleration": 50000,
            "x": options.get("x"),
            "y": options.get("y"),
            "width": 10,
            "height": 70,
            "angle": options.get("angle"),
            "damage": 100,
            "range": 50000,
            "fireRate": 0.005,
            "energyCost": 1000,
            "penetration": 5,
            "ship": options.get("ship"),
            "color": options.get("color"),
            "img": options.get("img"),
        }
        super().__init__(defaults, force)
